torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1926,18 +1938,26 @@ def hint():
                 if r.upper != sympy.oo and r.upper < sys.maxsize - 1:
                     bounds.append(str(r.upper))
                 if len(bounds) > 1:
-                    exprs.append(" <= ".join(bounds))
+                    exprs.append(ShapeGuardExprSources(" <= ".join(bounds), sources))
 
         if constraint_violations:
-            msgs = [f"  {i + 1}. {msg()}" for i, msg in enumerate(constraint_violations)]
-            msgs = "\n".join(msgs)
-            raise ConstraintViolationError(f"Constraints violated!\n{msgs}")
+            # TODO: Is this sound? On one hand, it looks like "no". On the other hand, if we are in automatic
```

Comment:
@ezyang please opine here

Reply:
Naively, what I would have expected is you program the constraints into shape env differently for automatically found dynamic operations versus not, so that you never report violations. However, it is an interesting point that if we observe something varies, but we can't compile it dynamic, this is a notable even worth reporting (but we shouldn't error, as you did here.)

Reply:
Yes, I wanted to keep reporting around - I initially was playing with a shape_env flag/produce_guards flag that changed behavior, but I opted for the intersection of collecting violations but not raising.

Reply:
Toggling it globally is still the wrong thing to do. You could have regular `mark_dynamic` constraints which do need to raise an error. One way to structure this is to add a superclass to the constraints with `warn_only: bool`, and then filter constraints into errors vs warnings.